crypto/fipsmodule/sha/sha256.c

```diff
@@ -169,18 +164,11 @@ static int sha256_final_impl(uint8_t *out, SHA256_CTX *c) {
 }
 
 int SHA256_Final(uint8_t out[SHA256_DIGEST_LENGTH], SHA256_CTX *c) {
-  // Ideally we would assert |sha->md_len| is |SHA256_DIGEST_LENGTH| to match
-  // the size hint, but calling code often pairs |SHA224_Init| with
-  // |SHA256_Final| and expects |sha->md_len| to carry the size over.
-  //
-  // TODO(davidben): Add an assert and fix code to match them up.
-  return sha256_final_impl(out, c);
+  return sha256_final_impl(out, SHA256_DIGEST_LENGTH, c);
 }
+
 int SHA224_Final(uint8_t out[SHA224_DIGEST_LENGTH], SHA256_CTX *ctx) {
```

Comment:
Not sure why the warning isn't showing up for https://github.com/awslabs/aws-lc/blob/59eccdf959c93a4543b8cb4ae7af01e7be265d3e/crypto/fipsmodule/sha/sha512.c#L184-L188 since it's very similar, but this fix doesn't seem to break anything and adds extra assertions.

Reply:
The warning reported on `sha256.c` does not make sense at all. Not sure how to understand the error message given the code

https://github.com/awslabs/aws-lc/blob/59eccdf959c93a4543b8cb4ae7af01e7be265d3e/crypto/fipsmodule/sha/sha256.c#L161-L166